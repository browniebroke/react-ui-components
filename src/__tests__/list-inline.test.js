import React from 'react'
import { ThemeProvider } from 'styled-components'
import renderer from 'react-test-renderer'

import ListInline from '../list-inline'
import styledComponentsTheme from '../styled-components-theme'

describe('ListInline component', () => {
  test('should render correctly with children', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <ListInline>
            <div>This</div>
            <div>is</div>
            <div>an</div>
            <div>example</div>
          </ListInline>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly without children', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <ListInline />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render with custom padding', () => {
    let tree = renderer
      .create(
        <ThemeProvider theme={styledComponentsTheme}>
          <ListInline padding={0}>
            <div>This</div>
            <div>is</div>
            <div>fun</div>
          </ListInline>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
