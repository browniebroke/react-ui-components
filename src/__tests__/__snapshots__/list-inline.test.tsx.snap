// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListInline component should render correctly with children 1`] = `
<ul
  className="sc-iBPRYJ ljKKDo"
>
  <li
    className="sc-fubCfw hUayPB"
  >
    <div>
      This
    </div>
  </li>
  <li
    className="sc-fubCfw hUayPB"
  >
    <div>
      is
    </div>
  </li>
  <li
    className="sc-fubCfw hUayPB"
  >
    <div>
      an
    </div>
  </li>
  <li
    className="sc-fubCfw hUayPB"
  >
    <div>
      example
    </div>
  </li>
</ul>
`;

exports[`ListInline component should render correctly without children 1`] = `
<ul
  className="sc-iBPRYJ ljKKDo"
/>
`;

exports[`ListInline component should render with custom padding 1`] = `
<ul
  className="sc-iBPRYJ ljKKDo"
>
  <li
    className="sc-fubCfw tKMDB"
  >
    <div>
      This
    </div>
  </li>
  <li
    className="sc-fubCfw tKMDB"
  >
    <div>
      is
    </div>
  </li>
  <li
    className="sc-fubCfw tKMDB"
  >
    <div>
      fun
    </div>
  </li>
</ul>
`;
